### ISSUE TITLE
java.lang.ClassNotFoundException: net.minecraft.text.Text on Fabric 1.20.1

### ISSUE THREAD

**Issue by burdoto** [Mineaurion/AurionChat#25]
Hello, I am encountering an issue using this plugin on a Fabric 1.20.1 server.
Looking at the files, you seem to target this specific version; however I couldn't get it to work even by creating a custom build.

Here's the full log: https://hastebin.com/share/qicaluheva.sql

**Comment by Yann151924** [comment]
Hi,
With the latest version normally it work with the 1.20.1 fabric, if not i can release a new one.
If you want to custom build you need to change this : https://github.com/Mineaurion/Aurionchat/blob/master/gradle.properties#L3 with like "1.0.0" forge and fabric don't like the "SNAPSHOT" word in the version

**Comment by burdoto** [comment]
i didnt have issues with the version name, but thats a good note thanks

however, it will not load for some reason.
similarly, the bukkit version also doesnt initialize (inside an arclight instance):
https://hastebin.com/share/efavocuhoq.markdown
rabbitmq url was checked multiple times, that one is alright

**Comment by Gaetann18** [comment]
did you correctly specify the port in the rabbit mq url?

**Comment by burdoto** [comment]
yes, it was checked several times and uses the correct port, ip, vhost, username and password

**Comment by Gaetann18** [comment]
can you attach your configuration file to me?

**Comment by burdoto** [comment]
https://pastebin.com/XZjXrtPL

**Comment by Gaetann18** [comment]
rabbitmq:
  uri: "amqp://minecraft:youpassword@yourip:5672"
remove /minecraft

**Comment by burdoto** [comment]
rabbitmq docs state that [the URI should be terminated with a segment that defines the Vhost to be used](https://www.rabbitmq.com/uri-spec.html), but i will try that in a minute

**Comment by Gaetann18** [comment]
Of memory in the plugin code it will automatically increment

**Comment by burdoto** [comment]
has a similar result without the vhost parameter: https://pastebin.com/aW5RPenj

**Comment by Gaetann18** [comment]
use this template and everything should work

``` 
rabbitmq:
  uri: "amqp://youruser:yourpassword@yourip:5672"
  servername: "youservername1" #Must be unique beetwen all of our servers 


options:
  spy: false
  automessage: false
  sound: "ENTITY_EXPERIENCE_ORB_PICKUP"

channels:
   youservername1:
     format: "[&a PREFIX &f] {prefix}{display_name} : &f{message}"
     alias: "youralias"
   youservername2:
     format: "[&6PREFIX &f] {prefix}{display_name} : &f{message}"
     alias: "yoursecondalias"
```

**Comment by burdoto** [comment]
that made a difference; now initialization doesn't fail, but handling chat messages fails https://pastebin.com/D17bRpuJ

**Comment by Gaetann18** [comment]
for me the error comes from the fact that aurionchat does not reach this co at rabbitmq. either a user rights problem or something else but it doesn't come from the plugin for me hum

**Comment by burdoto** [comment]
admittedly, i'm kinda new to rabbitmq; but i did follow docs in order to give the minecraft user full access to the minecraft vhost

**Comment by Gaetann18** [comment]
> that made a difference; now initialization doesn't fail, but handling chat messages fails https://pastebin.com/D17bRpuJ

you have luckperms on this server ?

**Comment by burdoto** [comment]
yes, also the bukkit version. v5.4.40

i want to note that initialization only worked when specifying the `minecraft` vhost in the URI
when removing that again, initialization fails again; https://pastebin.com/hzg5iGNR

**Comment by burdoto** [comment]
maybe you know of some simple (maybe web based?) rabbitmq implementation that i can use for testing permissions?

**Comment by Gaetann18** [comment]
> yes, also the bukkit version. v5.4.40
> 
> i want to note that initialization only worked when specifying the `minecraft` vhost in the URI when removing that again, initialization fails again; https://pastebin.com/hzg5iGNR

hum 

strange because here is one of our configurations that works from aurionchat


```rabbitmq:
  uri: "amqp://mineaurion:superpassword@rabbit.mineaurion.com:5672"
  servername: "ragnamod-arclight"

options:
  spy: false
  automessage: true
  sound: "ENTITY_EXPERIENCE_ORB_PICKUP"

channels:
   global:
     format: "[&aGlobal&f] {prefix}{display_name} : &f{message}"
     alias: "g"
   infinity:
     format: "[&6I&f] {prefix}{display_name} : &f{message}"
     alias: "inf"
   agrarian:
     format: "[&dAS&f] {prefix}{display_name} : &f{message}"
     alias: "agra"
   ragnamod-arclight:
     format: "[&eR6&f] {prefix}{display_name} : &f{message}"
     alias: "rag"
   infinityreloaded:
     format: "[&eIR&f] {prefix}{display_name} : &f{message}"
     alias: "infr"
   omnifactory:
     format: "[&eOMF&f] {prefix}{display_name} : &f{message}"
     alias: "omf"
   atm7:
     format: "[&aATM7&f] {prefix}{display_name} : &f{message}"
     alias: "atm7"
   packdonateur:
     format: "[&6DP&f] {prefix}{display_name} : &f{message}"
     alias: "dp"
   blackstoneblock:
     format: "[&6BSB&f] {prefix}{display_name} : &f{message}"
     alias: "bsb"
   megalos:
     format: "[&bMGLS&f] {prefix}{display_name} : &f{message}"
     alias: "mega"```

**Comment by Gaetann18** [comment]
> maybe you know of some simple (maybe web based?) rabbitmq implementation that i can use for testing permissions?


on the web config part of rabbit mq only @Yann151924 will be able to help you

**Comment by burdoto** [comment]
could it be that your rabbitmq impl is fetching Vhost based on connection parameters? (domain, servername, etc?)
as mentioned this is my first time using rabbitmq so it could as well be a configuration issue.
i am not aware of way to test rabbitmq

**Comment by burdoto** [comment]
huh! i didnt even know there was a web config, this far i've been using the `rabbitmqctl` tool

**Comment by burdoto** [comment]
ok well, lets assume for bukkit the issue is my configuration; that still doesnt solve the actual issue with the mod for fabric; i still cannot start it at all for the same exception as before

**Comment by Gaetann18** [comment]
> ok well, lets assume for bukkit the issue is my configuration; that still doesnt solve the actual issue with the mod for fabric; i still cannot start it at all for the same exception as before

you have the same connection error with fabric?

**Comment by burdoto** [comment]
on fabric; i get this exception on startup: https://hastebin.com/share/qicaluheva.sql

**Comment by Gaetann18** [comment]
> on fabric; i get this exception on startup: https://hastebin.com/share/qicaluheva.sql

did you try to compile the mod with what Yann told you to modify?

**Comment by burdoto** [comment]
this occurred with the [prebuilt variant from the releases page](https://github.com/Mineaurion/Aurionchat/releases/tag/v0.14.3) which denotes version `0.14.3`;
also occurred inside the IDE before building, 
also occurred with a custom shadow build without the adjustment

**Comment by Gaetann18** [comment]
the version of the fabric loader between your server and the plugin is different, maybe that's the reason. Try downgrading your server or updating here
https://github.com/Mineaurion/Aurionchat/blob/e00617de47311d01dc46e2283ade00a23b624b56/fabric/build.gradle#L16C1-L16C1

**Comment by burdoto** [comment]
not sure if we can downgrade on the server; HOWEVER that doesnt solve the issue; I'm still getting the same ClassNotFoundException

**Comment by Gaetann18** [comment]
try this: I updated the dependencies in the fabric build.gradle (not tested on a server) : 

[AurionChat-fabric-latest-SNAPSHOT-all.zip](https://github.com/Mineaurion/Aurionchat/files/13804363/AurionChat-fabric-latest-SNAPSHOT-all.zip)

**Comment by burdoto** [comment]
thanks for your patience and work at this point!
but unfortunately, this STILL yields the same ClassNotFoundException :/

**Comment by Gaetann18** [comment]
try "Fingers crossed"
[AurionChat-fabric-latest-99-all.zip](https://github.com/Mineaurion/Aurionchat/files/13804377/AurionChat-fabric-latest-99-all.zip)

**Comment by burdoto** [comment]
still the same 😬

**Comment by Yann151924** [comment]
With rabbitmq you can have a web ui to have better understanding of what you are doing : https://www.rabbitmq.com/management.html
Really usefull if you need to see if any queue,exchange etc is declared.

You can put the vhost after the / or not depending on what you want but the user you are using need to have enough permission on the vhost. For testing purpose try the admin ("root") user but after you need to test some permission and look at your rabbitmq log.

For the "ClassNotFoundException" error, i need more log i don't see any reference to our "mod/plugin" it could be another mod doing this at this point.


For the another error in this pastebin : https://pastebin.com/raw/D17bRpuJ . I see you are using plugman : 
```
[10:49:09 INFO]: Kaleidox issued server command: /plugman load AurionChat-bukkit-latest-0.14.3-all
```

Please don't use this, the plugin will break and have unexpected issue. Just reboot your server it's better, some stuff in the plugin happen when the server is loaded

**Comment by Gaetann18** [comment]
I have a doubt about the java version. are you running java 17?

**Comment by Yann151924** [comment]
> still the same 😬

If you can provide a quick setup to reproduce the error, i'm happy to test it locally on my pc.

**Comment by burdoto** [comment]
this is java 19;
```
openjdk 19.0.2 2023-01-17
OpenJDK Runtime Environment (build 19.0.2+7-44)
OpenJDK 64-Bit Server VM (build 19.0.2+7-44, mixed mode, sharing)
```
however, this was never an issue before; and manually using java 17 doesnt yield any different result


about plugman; yes i realize that it can make a breaking difference; and do notice that happening from time to time. i have tried without and a new issue has arose;
chat is now entirely broken! :D only the /minecraft:say command still works, everything else is being swallowed
i'm starting to feel like in the bukkit scenario, after normally restarting this is an issue of listener priority, since we use LPC for chat formatting as it supports PlaceholderAPI


reproducing this to me was as simple as 
- downloading a 1.20.1 fabric server jar; be it from serverjars.org or fabric themselves
- adding AurionChat and starting once for a blank config file
- configurating
- starting again; has been happening ever since and server couldn't start

**Comment by burdoto** [comment]
i've been able to look into the rabbitmq web config; and it seems like everything is fine here; the minecraft user does have access to the minecraft vhost.
am i maybe missing channel- or topic- configuration?

**Comment by Yann151924** [comment]
Hum yes aurionchat his top priority and override anything else. This is a part i need to change or adapt especially in case of rabbitmq outage. 

You need at least `aurionchat.chat.speak` to at least talk.

> i've been able to look into the rabbitmq web config; and it seems like everything is fine here; the minecraft user does have access to the minecraft vhost. am i maybe missing channel- or topic- configuration?

Exchange and queue are automaticly created if nothing appear you need to check your rabbitmq log something happen with your user cannot create this it think.

**Comment by burdoto** [comment]
so, i see the aurion.chat exchange on the rabbitmq server

however i was kinda wrong about the reproduction;
i cant start fabric server with it at all; i confused that with the bukkit one.
bukkit one i could get to work now, however its a big issue that we cannot use placeholderapi anymore, do you think you could get that in as well with the new fabric release?

also, we would like to post the servername that a message came from within the message format; but i am unsure of the replacement variables

**Comment by Yann151924** [comment]
I need some time to implement placeholderapi and make it work with the config file (not happening in next weeks)

Can you try this test version of fabric. I build it with the latest 1.20.1 version of fabric, with a blank setup it work for me
[AurionChat-fabric-latest-1.0.0.zip](https://github.com/Mineaurion/Aurionchat/files/13804709/AurionChat-fabric-latest-1.0.0.zip)

I never test this but the config is used when you send message. So theoretically you can add like `[SRV-A]` in the config for server A and `[SRV-B]` in the config of server b and you can distinguish two server.

And the `servername` key in the config @Gaetann18 gave you is obsolete. I removed in the earlier version.

**Comment by burdoto** [comment]
could you specify your exact fabric server jar, as i'm still getting this very same issue?

yes, i realized as well that i could just specify the source server name from the message format
however; colors are a bit funny; they require the MOTD notation

**Comment by burdoto** [comment]
oh wait i'm an idiot and confused all the different versions that i had downloaded

this file you provided works now 😄 
thank you very much for your effort in resolving this

**Comment by Yann151924** [comment]
I'll make a new version this afternoon or next few days fixing this.
For the color, it's weird but i have not tested very well
